core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java

```diff
@@ -0,0 +1,745 @@
+                    return false;
+                }
+
+                return true;
+            }
+
+            if (event.rebalanceFromThreadId == id) {
+                Parser parser = parserCache.get(event.rebalanceTableName);
+                parserCache.remove(event.rebalanceTableName);
+                parser.close();
+                event.rebalanceReleasedByFromThread = true;
+                return true;
+            }
+
+            return true;
+        }
+
+        private class Parser implements Closeable {
+            private TableWriter writer;
+            private final IntList colTypes = new IntList();
+            private final IntList colIndexMappings = new IntList();
+
+            private transient int nMeasurementValues;
+            private transient boolean error;
+
+            private void processFirstEvent(CairoEngine engine, CairoSecurityContext securityContext, LineTcpMeasurementEvent event) {
+                assert null == writer;
+                int status = engine.getStatus(securityContext, path, event.getTableName(), 0, event.getTableName().length());
+                if (status == TableUtils.TABLE_EXISTS) {
+                    writer = engine.getWriter(securityContext, event.getTableName());
+                    processEvent(event);
+                    return;
+                }
+
+                preprocessEvent(event);
+                engine.creatTable(
+                        securityContext,
+                        appendMemory,
+                        path,
+                        tableStructureAdapter.of(event, this));
+                int nValues = event.getNValues();
+                for (int n = 0; n < nValues; n++) {
+                    colIndexMappings.add(n, n);
+                }
+                writer = engine.getWriter(securityContext, event.getTableName());
+                addRow(event);
+            }
+
+            private void processEvent(LineTcpMeasurementEvent event) {
+                assert event.getTableName().equals(writer.getName());
+                preprocessEvent(event);
+                parseNames(event);
+                addRow(event);
+            }
+
+            private void addRow(LineTcpMeasurementEvent event) {
+                if (error) {
+                    return;
+                }
+                long timestamp = event.getTimestamp();
+                Row row = writer.newRow(timestamp);
+                try {
+                    for (int i = 0; i < nMeasurementValues; i++) {
+                        int columnType = colTypes.getQuick(i);
+                        int columnIndex = colIndexMappings.getQuick(i);
+                        CairoLineProtoParserSupport.writers.getQuick(columnType).write(row, columnIndex, event.getValue(i));
+                    }
+                    row.append();
+                } catch (BadCastException ignore) {
+                    row.cancel();
+                }
+                writer.commit();
```

Comment:
commit() rate should be configurable to avoid committing every row. Commit can definitely be done when either X rows reached or queue becomes empty